yt/yt/tests/integration/node/test_job_proxy_log_manager.py

```diff
@@ -0,0 +1,159 @@
+            "resource_limits": {
+                "user_slots": 5,
+                "cpu": 5,
+                "memory": 5 * 1024 ** 3,
+            }
+        },
+    }
+
+    @authors("tagirhamitov")
+    def test_removing_logs(self):
+        job_count = 3
+
+        op = run_test_vanilla(with_breakpoint("BREAKPOINT"), job_count=job_count)
+        job_ids = wait_breakpoint(job_count=job_count)
+        assert len(job_ids) == job_count
+        release_breakpoint()
+
+        op.track()
+
+        for job_id in job_ids:
+            assert self.job_proxy_log_exists(job_id)
+
+        time.sleep(1.2)
+
+        for job_id in job_ids:
+            assert not self.job_proxy_log_exists(job_id)
+
+    @authors("tagirhamitov")
+    def test_removing_logs_on_start(self):
+        job_count = 3
+
+        op = run_test_vanilla(with_breakpoint("BREAKPOINT"), job_count=job_count)
+        job_ids = wait_breakpoint(job_count=job_count)
+        assert len(job_ids) == job_count
+        release_breakpoint()
+
+        op.track()
+
+        for job_id in job_ids:
+            assert self.job_proxy_log_exists(job_id)
+
+        with Restarter(self.Env, NODES_SERVICE):
+            time.sleep(1.2)
+
+        time.sleep(0.2)
+
+        for job_id in job_ids:
+            assert not self.job_proxy_log_exists(job_id)
+
+
+class TestJobProxyLogManagerDynamicConfig(TestJobProxyLogManagerBase):
+    DELTA_NODE_CONFIG = {
+        "exec_node": {
+            "job_controller": {
+                "resource_limits": {
+                    "user_slots": 5,
+                    "cpu": 5,
+                    "memory": 5 * 1024 ** 3,
+                }
+            },
+            "job_proxy": {
+                "job_proxy_logging": {
+                    "mode": "per_job_directory",
+                },
+            },
+            "job_proxy_log_manager": {
+                "logs_storage_period": "60s",
+            },
+        },
+        "job_resource_manager": {
+            "resource_limits": {
+                "user_slots": 5,
+                "cpu": 5,
+                "memory": 5 * 1024 ** 3,
+            }
+        },
+    }
+
+    @authors("tagirhamitov")
+    def test_dynamic_config(self):
+        op = run_test_vanilla(with_breakpoint("BREAKPOINT"))
+        job_id_1 = wait_breakpoint()[0]
+        release_breakpoint(job_id=job_id_1)
+        op.track()
+
+        assert self.job_proxy_log_exists(job_id_1)
+
+        update_nodes_dynamic_config({
+            "exec_node": {
+                "job_controller": {
+                    "job_proxy_log_manager": {
+                        "logs_storage_period": "1s",
+                    },
+                }
+            }
+        })
+
+        op = run_test_vanilla(with_breakpoint("BREAKPOINT"))
+        job_id_2 = wait_breakpoint()[0]
+        release_breakpoint(job_id=job_id_2)
+        op.track()
+
+        assert self.job_proxy_log_exists(job_id_2)
+
+        time.sleep(1.2)
```

Comment:
Let make here sleep on exactly 1 second, and for the second check below let use `wait`

Reply:
Maybe we can completely remove time.sleep here and instead wait for second log to be deleted and then check that the first log still exists?